### PR TITLE
Trusty upgrade script

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
tests: manual test for trusty upgrade to xenial

Add manual upgrade test to address trusty to xenial upgrade. This test will cover #1590 

## Test Steps
Run the manual test added here

## Desired commit type::

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: sru/release-27/test_trusty_upgrade_to_xenial.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# not enable do-release-upgrade to run a post script that handles the services
+# migration when performing an upgrade.
+# This script is adding a solution for that problem by manually adding the necessary
+# configs for do-release-upgrade (This configurations are done for release 26 forward)
+set -x
+
+series=trusty
+name=$series-upgrade-lxd
+
+function fix_upgrade_for_esm_infra() {
```
Comment (blackboxsw) [inline sru/release-27/test_trusty_upgrade_to_xenial.sh:15]: Let's document this issue URL here that we are working around

File: sru/release-27/test_trusty_upgrade_to_xenial.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+multipass delete $name
+multipass purge
+multipass launch $series --name $name
+
+multipass exec $name -- sudo apt-get upgrade -y
+multipass exec $name -- sudo apt-get dist-upgrade -y
+multipass exec $name -- ua version
+multipass exec $name -- sudo ua attach $UACLIENT_BEHAVE_CONTRACT_TOKEN
+multipass exec $name -- sudo ua status
+multipass exec $name -- sudo add-apt-repository ppa:orndorffgrant/uaclient-staging-27 -y
```
Comment (blackboxsw) [inline sru/release-27/test_trusty_upgrade_to_xenial.sh:30]: I think we should change this to actually setup -proposed on the system under test

Something like what we do for cloud-init (but installing ubuntu-advantage-tools instead) https://github.com/cloud-init/ubuntu-sru/blob/main/sru-scripts/sru-vars.template#L59-L76

**Review by blackboxsw — CHANGES_REQUESTED** [review]
thanks lucas, let's instrument something that tests the -proposed pocket if we can

**Review by blackboxsw — APPROVED** [review]
This looks good  thanks for the update.

File: sru/release-27/test_xenial_upgrade.sh [hunk 3]
```diff
@@ -65,7 +65,8 @@ function teardown_vm() {
 function install_new_ua() {
     tool=$1
     h2 "Set up to use daily ppa"
-    $tool exec $name -- sudo add-apt-repository ppa:ua-client/daily -y
+    $tool exec $name -- sudo sh -c "cat <<EOF >/etc/apt/sources.list.d/ubuntu-$series-proposed.list
```
Comment (orndorffgrant) [inline sru/release-27/test_xenial_upgrade.sh:68]: @lucasmoura Really sorry I should've mentioned - this script doesn't currently have `$series` set. Can you set `series=xenial` at the top of the script?

**Review by orndorffgrant — APPROVED** [review]

**Pull request merged**